test/Plutip.purs

```diff
@@ -0,0 +1,339 @@
+        -- Nothing is expected, because we are in an empty chain.
+        -- This test only checks for ability to connect to ODC
+        logInfo' <<< show =<< getDatumByHash
+          ( mkDatumHash
+              "42be572a6d9a8a2ec0df04f14b0d4fcbe4a7517d74975dfff914514f12316252"
+          )
+        logInfo' <<< show =<< getDatumsByHashes
+          [ mkDatumHash
+              "777093fe6dfffdb3bd2033ad71745f5e2319589e36be4bc9c8cca65ac2bfeb8f"
+          , mkDatumHash
+              "e8cb7d18e81b0be160c114c563c020dcc7bf148a1994b73912db3ea1318d488b"
+          ]
+
+    test "runPlutipContract: MintsMultipleTokens" do
+      let
+        distribution :: Array BigInt
+        distribution =
+          [ BigInt.fromInt 5_000_000
+          , BigInt.fromInt 2_000_000_000
+          ]
+      runPlutipContract config distribution \alice -> do
+        withKeyWallet alice do
+          tn1 <- mkTokenName "Token with a long name"
+          tn2 <- mkTokenName "Token"
+          mp1 /\ cs1 <- mkCurrencySymbol mintingPolicyRdmrInt1
+          mp2 /\ cs2 <- mkCurrencySymbol mintingPolicyRdmrInt2
+          mp3 /\ cs3 <- mkCurrencySymbol mintingPolicyRdmrInt3
+
+          let
+            constraints :: Constraints.TxConstraints Void Void
+            constraints = mconcat
+              [ Constraints.mustMintValueWithRedeemer
+                  (Redeemer $ Integer (BigInt.fromInt 1))
+                  (Value.singleton cs1 tn1 one <> Value.singleton cs1 tn2 one)
+              , Constraints.mustMintValueWithRedeemer
+                  (Redeemer $ Integer (BigInt.fromInt 2))
+                  (Value.singleton cs2 tn1 one <> Value.singleton cs2 tn2 one)
+              , Constraints.mustMintValueWithRedeemer
+                  (Redeemer $ Integer (BigInt.fromInt 3))
+                  (Value.singleton cs3 tn1 one <> Value.singleton cs3 tn2 one)
+              ]
+
+            lookups :: Lookups.ScriptLookups Void
+            lookups =
+              Lookups.mintingPolicy mp1
+                <> Lookups.mintingPolicy mp2
+                <> Lookups.mintingPolicy mp3
+
+          ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+          bsTx <-
+            liftedM "Failed to balance/sign tx" $ balanceAndSignTx ubTx
+          submitAndLog bsTx
+
+    test "runPlutipContract: SignMultiple" do
+      let
+        distribution :: Array BigInt
+        distribution =
+          [ BigInt.fromInt 100_000_000
+          -- move this entry one position up in the list to reproduce the bug:
+          -- TODO:
+          -- https://github.com/Plutonomicon/cardano-transaction-lib/issues/668
+          , BigInt.fromInt 5_000_000
+          ]
+      runPlutipContract config distribution \alice -> do
+        withKeyWallet alice do
+          pkh <- liftedM "Failed to get own PKH" ownPaymentPubKeyHash
+          let
+            constraints :: Constraints.TxConstraints Void Void
+            constraints = Constraints.mustPayToPubKey pkh
```

Comment:
We can't provide a meaningful stake key (because plutip does not provide it). I will add a comment saying that in real examples we should use the longer variant)